### PR TITLE
Update Apalache version to `0.30.x`

### PR DESCRIPTION
**Pull request by rnbguy**
- [x] Support extra args to pass to Apalache.
  - Specifically `--features=no-rows` flag for back compatibility with type system 1. 
- [x] Merge, release and use informalsystems/modelator-py#66

### CODE REVIEW

**Review by ivan-gavran — APPROVED** [review]
Looks good, modulo three small comments I left.

One major thing that I was missing is an automatic testing of the introduced changes, so I added a basic test. (I think its value is as an instruction how to use for newcomers, rather than as an integration test)

File: modelator/utils/modelator_helpers.py [hunk 1]
```diff
@@ -44,9 +44,6 @@ def wrap_command(
     json_command = {}
     json_command["args"] = {}
```
[the remark this answers is not in the document]
Comment (rnbguy) [reply modelator/utils/modelator_helpers.py:46]: I moved it down.

File: modelator/utils/modelator_helpers.py [hunk 2]
```diff
@@ -74,11 +66,37 @@ def wrap_command(
...
+
+    # send only basenames of files to modelator-py
+    json_command["files"] = {os.path.basename(f): files[f] for f in files}
+
     if checker == const_values.TLC:
         json_command["jar"] = os.path.abspath(const_values.DEFAULT_TLC_JAR)
     else:
         json_command["jar"] = os.path.abspath(const_values.DEFAULT_APALACHE_JAR)
 
+    if checker == const_values.APALACHE:
```
Comment (rnbguy) [inline modelator/utils/modelator_helpers.py:97]: @ivan-gavran specifically here.

File: tests/test_model_check.py [hunk 3]
```diff
@@ -35,11 +35,13 @@ def _matching_check_value(
     assert check_result.is_ok == expected_result
     assert (check_result.error_msg is None) == check_result.is_ok
 
-    assert len(check_result.trace_paths) == 1
     trace_filenames = [os.path.basename(p) for p in check_result.trace_paths]
     if check_result.is_ok:
+        assert len(check_result.trace_paths) == 0
+        # doesn't make sense
```
[the remark this answers is not in the document]
Comment (rnbguy) [reply tests/test_model_check.py:41]: hmm very good catch. I don't remember putting it here :sweat_smile: I will remove it.

File: modelator/utils/modelator_helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivan-gavran) [reply modelator/utils/modelator_helpers.py:46]: aha, I missed that. good!

File: modelator/utils/modelator_helpers.py [hunk 4]
```diff
@@ -74,11 +66,37 @@ def wrap_command(
         for arg in args:
             json_command["args"][arg] = args[arg]
 
+    if cmd == const_values.PARSE_CMD:
+        for e in [
```
[the remark this answers is not in the document]
Comment (rnbguy) [reply modelator/utils/modelator_helpers.py:70]: hmm. I agree. but you also pointed out a very valid issue. I shouldn't modify an iterator while iterating.

**Comment by rnbguy** [comment]
@ivan-gavran @hvanz do you know why `modelator` is nondeterministically failing with exit code 5 on macOS?

**Comment by rnbguy** [comment]
I am merging this PR as I am unsure how to debug the situation at macOS runner. I added a [tmate action](https://github.com/mxschmitt/action-tmate) if it fails to log into the GitHub workflow via ssh.

**Pull request merged**